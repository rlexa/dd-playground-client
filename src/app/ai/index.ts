export * from './ml-polynomial';
export * from './ml-polynomial-model';
export * from './tfjs.service';
export * from './util';
