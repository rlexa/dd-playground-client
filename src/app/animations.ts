import {animate, keyframes, style, transition, trigger} from '@angular/animations';

const ANIM_ENTER_APPEAR = [style({opacity: '0'}), animate('150ms ease-in')];
const ANIM_ENTER_LEFT = [style({transform: 'translateX(50%)', opacity: '0'}), animate('200ms ease-in')];
const ANIM_LEAVE_BOTTOM = [animate('150ms ease-out', style({transform: 'translateY(50%)', opacity: '0'}))];
const ANIM_LEAVE_DISAPPEAR = [animate('150ms ease-out'), style({opacity: '0'})];
const ANIM_ROTATE_X = [style({transform: 'rotateX(-360deg)'}), animate('600ms 50ms ease-in')];
const ANIM_SCALE_UP_DOWN_UP_DOWN = [
  animate(
    '600ms 50ms ease-out',
    keyframes([
      style({offset: 0, transform: 'scale(1)'}),
      style({offset: 0.25, transform: 'scale(1.1)'}),
      style({offset: 0.5, transform: 'scale(1)'}),
      style({offset: 0.75, transform: 'scale(1.05)'}),
      style({offset: 1, transform: 'scale(1)'}),
    ]),
  ),
];
const ANIM_SCALE_BIG_UP_DOWN_UP_DOWN = [
  animate(
    '600ms 50ms ease-out',
    keyframes([
      style({offset: 0, transform: 'scale(1)'}),
      style({offset: 0.25, transform: 'scale(1.75)'}),
      style({offset: 0.5, transform: 'scale(1.25)'}),
      style({offset: 0.75, transform: 'scale(1.5)'}),
      style({offset: 1, transform: 'scale(1)'}),
    ]),
  ),
];
const ANIM_WOBBLE_X = [
  animate(
    '400ms 50ms ease-out',
    keyframes([
      style({offset: 0, transform: 'translateX(0rem)'}),
      style({offset: 0.25, transform: 'translateX(.3rem)'}),
      style({offset: 0.5, transform: 'translateX(0rem)'}),
      style({offset: 0.75, transform: 'translateX(-.3rem)'}),
      style({offset: 1, transform: 'translateX(0rem)'}),
    ]),
  ),
];

// Triggers

export const TRIGGER_ENTER_APPEAR = trigger('TRIGGER_EnterAppear', [transition(':enter', [...ANIM_ENTER_APPEAR])]);
export const TRIGGER_ENTER_APPEAR_LEAVE_DISAPPEAR = trigger('TRIGGER_EnterAppear_LeaveDisappear', [
  transition(':enter', [...ANIM_ENTER_APPEAR]),
  transition(':leave', [...ANIM_LEAVE_DISAPPEAR]),
]);
export const TRIGGER_ENTER_LEFT = trigger('TRIGGER_EnterLeft', [transition(':enter', [...ANIM_ENTER_LEFT])]);
export const TRIGGER_ENTER_LEFT_LEAVE_BOTTOM = trigger('TRIGGER_EnterLeft_LeaveBottom', [
  transition(':enter', [...ANIM_ENTER_LEFT]),
  transition(':leave', [...ANIM_LEAVE_BOTTOM]),
]);
export const TRIGGER_ENTER_SCALE_BIG_UP_DOWN_UP_DOWN = trigger('TRIGGER_EnterScaleBigUpDownUpDown', [
  transition(':enter', [...ANIM_SCALE_BIG_UP_DOWN_UP_DOWN]),
]);
export const TRIGGER_ROTATE_X = trigger('TRIGGER_RotateX', [transition('* <=> *', [...ANIM_ROTATE_X])]);
export const TRIGGER_SCALE_UP_DOWN_UP_DOWN = trigger('TRIGGER_ScaleUpDownUpDown', [transition('* <=> *', [...ANIM_SCALE_UP_DOWN_UP_DOWN])]);
export const TRIGGER_WOBBLE_X = trigger('TRIGGER_WobbleX', [transition('* <=> *', [...ANIM_WOBBLE_X])]);
