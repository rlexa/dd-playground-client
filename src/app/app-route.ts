export enum AppRoute {
  Dashboard = 'dashboard',
}
