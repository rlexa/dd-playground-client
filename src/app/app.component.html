<div class="position-absolute match-parent app-viewport">
  <div class="position-absolute match-parent container-bg-logo"></div>
  <ng-container *ngIf="(isForcingStart$ | async) || isBrowserChrome; else templateNeedChrome">
    <router-outlet></router-outlet>
  </ng-container>
  <ng-template #templateNeedChrome>
    <div ddFlexbox="vxcyc" class="position-absolute match-parent">
      <h1 class="color-warn">Please switch to Chrome browser.</h1>
      <button *ngIf="!isBrowserIE" (click)="onForceStart()">Force</button>
    </div>
  </ng-template>
</div>