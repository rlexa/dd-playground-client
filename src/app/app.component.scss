@import '~styles.scss';
.app-viewport {
    min-width: $size-viewport-min;
}