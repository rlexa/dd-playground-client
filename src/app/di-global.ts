import {HttpClient} from '@angular/common/http';
import {inject, InjectionToken, Provider} from '@angular/core';
import {ActivatedRouteSnapshot, NavigationEnd, Router} from '@angular/router';
import {Observable, of} from 'rxjs';
import {catchError, distinctUntilChanged, filter, map, shareReplay, startWith} from 'rxjs/operators';
import {isEqualValue} from './util';

export interface GlobalFlags {
  buildId?: string;
  buildRevision?: string;
  buildSystem?: string;
  buildVariant?: string;
  project?: string;
  projectParent?: string;
  title?: string;
  version?: string;
}

/** Global flags. */
export const DiGlobalFlags = new InjectionToken<Observable<GlobalFlags>>('Global flags.', {
  providedIn: 'root',
  factory: () =>
    inject(HttpClient)
      .get<GlobalFlags>('/assets/flags.json', {responseType: 'json'})
      .pipe(
        catchError(() => of<GlobalFlags>(null)),
        distinctUntilChanged(isEqualValue),
        shareReplay(1),
      ),
});

/** Global title. */
export const DiGlobalTitle = new InjectionToken<Observable<string>>('Global title.', {
  providedIn: 'root',
  factory: () =>
    inject(DiGlobalFlags).pipe(
      map((flags) => flags?.title),
      distinctUntilChanged(),
      shareReplay(1),
    ),
});

/** Global version. */
export const DiGlobalVersion = new InjectionToken<Observable<string>>('Global version.', {
  providedIn: 'root',
  factory: () =>
    inject(DiGlobalFlags).pipe(
      map((flags) => flags?.version),
      distinctUntilChanged(),
      shareReplay(1),
    ),
});

const mergeParams = (route: ActivatedRouteSnapshot, into: Record<string, string> = {}) => {
  route.paramMap.keys.forEach((key) => (into[key] = route.paramMap.get(key)));
  if (route.children.length) {
    route.children.forEach((child) => mergeParams(child, into));
  }
  return into;
};

/** Global version. */
export const DiGlobalRouterParams = new InjectionToken<Observable<Record<string, string>>>('Global Router params.');
export const DiGlobalRouterParamsProvider: Provider = {
  provide: DiGlobalRouterParams,
  deps: [Router],
  useFactory: (router: Router) =>
    router.events.pipe(
      // events is a cold stream, let's start with a fake navigation event with current state's url
      startWith(new NavigationEnd(0, router.routerState.snapshot.url, router.routerState.snapshot.url)),
      // only need end of navigation events for the current final resolved route
      filter((ev) => ev instanceof NavigationEnd),
      map(() => mergeParams(router.routerState.snapshot.root)),
      distinctUntilChanged(isEqualValue),
      shareReplay(1),
    ),
};
