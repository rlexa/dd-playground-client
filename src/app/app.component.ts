import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
import { DateAdapter, NativeDateAdapter } from '@angular/material';
import { Title } from '@angular/platform-browser';
import { NavigationEnd, Router } from '@angular/router';
import { ReduxService, ReduxSetGlobalService } from 'app/redux';
import { BehaviorSubject } from 'rxjs';
import { filter } from 'rxjs/operators';

const userAgent = !window ? '' : window.navigator.userAgent;
const isBrowserChrome = userAgent.indexOf('Chrome') >= 0;
const isBrowserIE = userAgent.indexOf('MSIE') >= 0 || userAgent.match(/Trident.*rv\:11\./);

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent implements OnDestroy {
  constructor(
    title: Title,
    dateAdapter: DateAdapter<NativeDateAdapter>,
    router: Router,
    redux: ReduxService,
    reduxSet: ReduxSetGlobalService,
  ) {
    this.start = () => {
      title.setTitle(redux.state.globalValues.flags.title);
      dateAdapter.setLocale('de-DE');
      router.events.pipe(filter(ev => ev instanceof NavigationEnd)).subscribe(ev => reduxSet.setRoute(router.routerState.snapshot.url));
    };
    if (this.isBrowserChrome) {
      this.start();
    }
  }

  private start: () => void = null;

  readonly isBrowserChrome = isBrowserChrome;
  readonly isBrowserIE = isBrowserIE;
  readonly isForcingStart$ = new BehaviorSubject(false);

  ngOnDestroy() {
    this.isForcingStart$.complete();
  }

  onForceStart() {
    this.isForcingStart$.next(true);
    this.start();
  }

}
