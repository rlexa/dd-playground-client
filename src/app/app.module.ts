import {HttpClientModule} from '@angular/common/http';
import {NgModule} from '@angular/core';
import {MatNativeDateModule, NativeDateModule} from '@angular/material/core';
import {BrowserModule} from '@angular/platform-browser';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {RouterModule, Routes} from '@angular/router';
import {AppRoute} from './app-route';
import {AppComponent} from './app.component';
import {DiGlobalRouterParamsProvider} from './di-global';
import {FlexboxModule} from './module/directive/flexbox';
import {ROUTE_ROOT, ROUTE_WILDCARD} from './routing';
import {AppRxStore, createAppRxStore} from './rx-state';

const appRoutes: Routes = [
  {
    path: AppRoute.Dashboard,
    loadChildren: () => import('src/app/module/widget/dashboard/dashboard.module').then((m) => m.DashboardModule),
  },
  {path: ROUTE_ROOT, redirectTo: AppRoute.Dashboard, pathMatch: 'full'},
  {path: ROUTE_WILDCARD, redirectTo: AppRoute.Dashboard},
];

@NgModule({
  /* DECLARATIONS: components, directives, and pipes belonging to that module (declare only once in app) */
  declarations: [AppComponent],
  /* IMPORTS: Import modules with exported declarable classes referenced in this module's component templates */
  imports: [
    // angular
    BrowserModule,
    HttpClientModule,
    BrowserAnimationsModule,
    RouterModule,
    // 3rd party
    MatNativeDateModule,
    NativeDateModule,
    FlexboxModule,
    // routing
    RouterModule.forRoot(appRoutes, {enableTracing: false}),
  ],
  /* EXPORTS: Export declared classes that components in other modules are able to reference in their templates */
  exports: [],
  /* PROVIDERS: all providers in any module imported in root have application scope */
  providers: [{provide: AppRxStore, useFactory: createAppRxStore}, DiGlobalRouterParamsProvider],
  bootstrap: [AppComponent],
})
export class AppModule {}
