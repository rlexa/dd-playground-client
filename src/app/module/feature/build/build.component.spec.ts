import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {of} from 'rxjs';
import {DiGlobalFlags} from 'src/app/di-global';
import {BuildComponent} from './build.component';

describe('BuildComponent', () => {
  let fixture: ComponentFixture<BuildComponent>;

  beforeEach(() => MockBuilder(BuildComponent).provide({provide: DiGlobalFlags, useValue: of('globalFlags')}));

  beforeEach(() => {
    fixture = MockRender(BuildComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
