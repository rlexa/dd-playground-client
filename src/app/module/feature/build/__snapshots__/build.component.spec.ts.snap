// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildComponent renders 1`] = `
<mock-render>
  <app-build>
    <mat-card
      appearance="outlined"
      ng-reflect-appearance="outlined"
    >
      <mat-card-title>
        Build Settings
      </mat-card-title>
      <mat-card-content>
        <app-simple-view
          ng-reflect-data="globalFlags"
          ng-reflect-is-width-constrained="false"
        />
      </mat-card-content>
    </mat-card>
  </app-build>
</mock-render>
`;
